modules/drivers/athena/src/metabase/driver/athena.clj

```diff
@@ -255,6 +257,18 @@
   [_driver _semantic-type expr]
   (hx/->time expr))
 
+(defmethod sql.qp/->honeysql [:athena :datetime-diff]
+  [driver [_ x y unit]]
+  (let [x (sql.qp/->honeysql driver x)
+        y (sql.qp/->honeysql driver y)]
+    (case unit
+      (:year :month :quarter :week :day)
+      (hsql/call :date_diff
+                 (hx/literal unit)
+                 (hsql/call :date_trunc (hx/literal :day) x)
+                 (hsql/call :date_trunc (hx/literal :day) y))
```

Comment:
Might be worth it to abstract out `date-trunc` into a separate helper function. Looks like it already is, but commented out, so maybe you can uncomment it and use that

Reply:
Updated https://github.com/metabase/metabase/pull/27205/commits/9d7c83cc1d870d6d61641a6aadfd14c2eb65c2db